src/platform/packages/shared/kbn-esql-ast/src/parser/cst_to_ast_converter.ts

```diff
@@ -573,10 +573,25 @@ export class CstToAstConverter {
     const commands: ast.ESQLCommand[] = [];
 
     if (fromCommandCtx) {
-      const fromCommand = this.fromFromCommand(fromCommandCtx);
-
-      if (fromCommand) {
-        commands.push(fromCommand);
+      // When parsing queries with nested empty subqueries like "FROM a, (FROM b, ())",
+      // ANTLR incorrectly identifies the empty parentheses "()" as a fromCommand context.
+      // This phantom context contains only closing parentheses (e.g., ")" or "))")
+      // without any actual FROM keyword, which would lead to malformed AST nodes.
+
+      // Solution: Verify that the context actually contains the "FROM" keyword before processing.
+      const fromCommandText = fromCommandCtx.getText();
+      const hasFromKeyword = /FROM/i.test(fromCommandText);
```

Comment:
Instead of `.getText()` and running regex on it: the `cst.FromCommandContext` has `.FROM()` field:

```ts
	public FROM(): TerminalNode {
		return this.getToken(esql_parser.FROM, 0);
	}
```

Maybe you can check that for `FROM` keyword existance?